src/analysis/souffle/sensitivity_analysis.dl

```diff
@@ -0,0 +1,134 @@
+#ifndef SRC_ANALYSIS_SOUFFLE_SENSITIVITY_ANALYSIS_DL_
+#define SRC_ANALYSIS_SOUFFLE_SENSITIVITY_ANALYSIS_DL_
+
+// this file defines the calculations for sensitivity analysis 
+// to be used in dp calculations
+#include "src/analysis/souffle/taint.dl"
+#include "src/analysis/souffle/sql_output.dl"
+
+#define CORE_INPUT_OPERATOR as("core.input", Operator)
+#define SQL_WHERE_OPERATOR as("sql.where", Operator)
+#define SQL_GROUP_BY_OPERATOR as("sql.group_by", Operator)
+#define SQL_SELECT_OPERATOR as("sql.select", Operator)
+#define SQL_FILTER_OPERATOR as("sql.filter", Operator)
+#define SQL_COLUMN_ACCESS_OPERATOR as("sql.column_access", Operator)
+#define SQL_OUTPUT_OPERATOR as("sql.output", Operator)
+#define SQL_SUM_OPERATOR as("sql.sum", Operator)
+#define SQL_COUNT_OPERATOR as("sql.count", Operator)
+#define SQL_MAX_OPERATOR as("sql.max", Operator)
+#define SQL_AVERAGE_OPERATOR as("sql.average", Operator)
+
+// One isSqlOperator fact per each sql operator, currently supporting one output.
+// We refer to results as 'value' (sqlValue declared below)
+.decl isSqlOperation(sqlOperation: Operation)
+isSqlOperation(operation) :- isOperation(operation), 
+      operationHasOperator(operation, operator), 
+      match("sql.*", operator). 
```

Comment:
For discovering a prefix, `match` is probably more power than necessary and potentially slow. I'd say use `substr` instead.

Reply:
Done.